crates/core_arch/src/mod.rs

```diff
@@ -264,7 +276,11 @@ mod arm;
 
 #[cfg(any(target_arch = "riscv32", target_arch = "riscv64", doc))]
 #[doc(cfg(any(target_arch = "riscv32", target_arch = "riscv64")))]
-mod riscv;
+mod riscv32;
```

Comment:
Perhaps calling this `riscv_shared` (like `arm_shared`) makes more sense?

Reply:
That makes sense, it would be a more common module system design.